tasks/linux.json

```diff
@@ -21,7 +21,7 @@
     }
   },
   "implementations": [
-    {"name": "linux.sh", "requirements": ["shell"], "files": ["package/files/common.sh"], "input_method": "environment"}
```

Comment:
I know you did not introduce this, but I feel like given this task does not have multiple implementations (similar to the powershell version), instead of using the `implementation` hash, instead we should just set `files: [...]` and `input_method`, instead of the entire implementations hash.

Reply:
@m0dular does this ^^ change make sense to you? Or do you not want to do it?

Reply:
I have to admit to not understanding how this works.  I remember some kind of issue where I had to add this stuff to `init.json`.  That currently has:

```
  "implementations": [
    {"name": "linux.sh", "requirements": ["shell"], "input_method": "environment", "files": ["package/files/common.sh", "package/files/apt.sh", "package/files/yum.sh", "package/files/zypper.sh"]}
  ],
```

Do we need to change that?  I changed `linux.json` like you said and it works, but I don't understand how the metadata impacts things.

Reply:
Ahh, sorry I was not clear. So when a task can have multiple implementations we use the implementations array. IN this case linux.sh is an implementation for the `package` (`init`) task. There is only a single implementation of linux.sh. Within an implementation hash you can set `files` and `input_method`. instead of setting them in an implementations array, you can set them both in the top level of this tasks metatdata. For example:
```
{
  "description": "Manage and inspect the state of packages (without a puppet agent)",
  "private": true,
  "input_method": "environment",
  "parameters": {
    "action": {
      "description": "The operation (install, status, uninstall and upgrade) to perform on the package.",
      "type": "Enum[install, status, uninstall, upgrade]"
    },
    "name": {
      "description": "The name of the package to be manipulated.",
      "type": "String[1]"
    },
    "version": {
      "description": "Version numbers must match the full version to install, including release if the provider uses a release moniker. Ranges or semver patterns are not accepted except for the gem package provider. For example, to install the bash package from the rpm bash-4.1.2-29.el6.x86_64.rpm, use the string '4.1.2-29.el6'.",
      "type": "Optional[String[1]]"
    },
    "provider": {
      "description": "The provider to use to manage or inspect the package, defaults to the system package manager. Only used when the 'puppet-agent' feature is available on the target so we can leverage Puppet.",
      "type": "Optional[String[1]]"
    }
  },
  "files": ["package/files/common.sh", "package/files/apt.sh", "package/files/yum.sh", "package/files/zypper.sh"]
}
```